Implement PartialEq between ArrayBase and ArrayRef

While working on updating `ndarray-npy` for `ndarray` 0.17.1, I noticed that the following doesn't compile:

```rust
use ndarray::{ArrayRef2, array};

fn main() {
    let arr = array![[1i32, 2, 3], [4, 5, 6]];
    let arr_ref: &ArrayRef2<i32> = &arr;
    assert_eq!(arr, arr_ref);
}
```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&code=use+ndarray%3A%3A%7BArrayRef2%2C+array%7D%3B%0A%0Afn+main%28%29+%7B%0A++++let+arr+%3D+array%21%5B%5B1i32%2C+2%2C+3%5D%2C+%5B4%2C+5%2C+6%5D%5D%3B%0A++++let+arr_ref%3A+%26ArrayRef2%3Ci32%3E+%3D+%26arr%3B%0A++++assert_eq%21%28arr%2C+arr_ref%29%3B%0A%7D%0A))

The error message is the following:

```
   Compiling playground v0.0.1 (/playground)
error[E0277]: can't compare `ArrayBase<OwnedRepr<i32>, Dim<[usize; 2]>, i32>` with `&ArrayRef<i32, Dim<[usize; 2]>>`
 --> src/main.rs:6:5
  |
6 |     assert_eq!(arr, arr_ref);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^ no implementation for `ArrayBase<OwnedRepr<i32>, Dim<[usize; 2]>, i32> == &ArrayRef<i32, Dim<[usize; 2]>>`
  |
  = help: the trait `PartialEq<&ArrayRef<i32, Dim<[usize; 2]>>>` is not implemented for `ArrayBase<OwnedRepr<i32>, Dim<[usize; 2]>, i32>`
  = note: this error originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider dereferencing both sides of the expression
 --> /playground/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/macros/mod.rs:46:22
  |
46|                 if !(**left_val == **right_val) {
  |                      +             +

For more information about this error, try `rustc --explain E0277`.
error: could not compile `playground` (bin "playground") due to 1 previous error
```

While there are several ways to work around this, it would be nice to make this example work as-is by adding the necessary `PartialEq` impl.

--- a/src/arraytraits.rs
+++ b/src/arraytraits.rs
@@ -717,6 +717,7 @@ where
 mod tests
 {
     use crate::array;
+    use alloc::vec;
 
     #[test]
     fn test_eq_traits()

